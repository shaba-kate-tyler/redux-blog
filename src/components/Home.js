import React from 'react';
import PropTypes from 'prop-types';
import Posts from './posts/Posts';

export default function Home({ posts }) {
  return <Posts posts={posts}/>;
}

Home.propTypes = {
  posts: PropTypes.array.isRequired
};
