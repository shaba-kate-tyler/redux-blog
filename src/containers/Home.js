import { withDisplayContent } from './higher-order/DisplayContent';
import { getTop5Posts } from '../selectors/posts';
import { fetchPosts } from '../actions/posts';
import Posts from '../components/posts/Posts';

export default withDisplayContent({
  contents: [
    {
      name: 'posts',
      getContent: getTop5Posts
    }
  ],
  fetches: [fetchPosts],
  Component: Posts
});
