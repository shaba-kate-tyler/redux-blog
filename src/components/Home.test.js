import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Home', () => {
  it('matches a snapshot', () => {
    const posts = [
      {
        id: '1',
        title: 'a post',
      },
      {
        id: '2',
        title: 'another post',
      },
      {
        id: '3',
        title: 'another other post',
      }
    ];
    const tree = renderer.create(
      <Home posts={posts}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
