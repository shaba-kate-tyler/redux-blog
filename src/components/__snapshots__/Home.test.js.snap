// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches a snapshot 1`] = `
<ul>
  <li>
    <a
      href="/posts/1"
      onClick={[Function]}
    >
      <h2>
        a post
      </h2>
    </a>
    <p>
      This is a post
    </p>
  </li>
  <li>
    <a
      href="/posts/2"
      onClick={[Function]}
    >
      <h2>
        another post
      </h2>
    </a>
    <p>
      This is a post
    </p>
  </li>
  <li>
    <a
      href="/posts/3"
      onClick={[Function]}
    >
      <h2>
        another other post
      </h2>
    </a>
    <p>
      This is a post
    </p>
  </li>
</ul>
`;
